apps/hash-ai-worker-ts/src/activities/flow-activities/research-entities-action/infer-entities-from-web-page-worker-agent.ts

```diff
@@ -832,26 +673,110 @@ export const inferEntitiesFromWebPageWorkerAgent = async (params: {
 
             const { outputs } = response.contents[0]!;
 
-            const newProposedEntities = outputs.find(
-              ({ outputName }) =>
-                outputName ===
-                ("proposedEntities" satisfies OutputNameForAction<"inferEntitiesFromContent">),
-            )?.payload.value as ProposedEntity[];
+            const editionProvenance: ProposedEntity["provenance"] = {
+              actorType: "ai",
+              // @ts-expect-error - `ProvidedEntityEditionProvenanceOrigin` is not being generated correctly from the Graph API
+              origin: {
+                type: "flow",
+                id: flowEntityId,
+                stepIds: [activityId],
+              } satisfies OriginProvenance,
+            };
+
+            const propertyProvenance: NonNullable<
+              ProposedEntity["propertyMetadata"]
+            >[number]["metadata"]["provenance"] =
+              "url" in toolCallInput
+                ? {
+                    sources: [
+                      {
+                        type: SourceType.Webpage,
+                        location: {
+                          uri: toolCallInput.url,
+                          /** @todo */
+                          name: undefined,
+                          description: undefined,
+                        },
+                        loadedAt: new Date().toISOString(),
+                        /** @todo */
+                        authors: undefined,
+                        firstPublished: undefined,
+                        lastUpdated: undefined,
+                      },
+                    ],
+                  }
+                : {
+                    sources: [
+                      {
+                        type: SourceType.Document,
+                        location: {
+                          uri: toolCallInput.fileUrl,
+                          /** @todo */
+                          name: undefined,
+                          description: undefined,
+                        },
+                        loadedAt: accessedRemoteFile?.loadedAt,
+                        /** @todo */
+                        authors: undefined,
+                        firstPublished: validAt,
```

Comment:
The document content may be valid but have been published earlier – this is especially true given that we assume a document is valid today if there is no information available on its validity (which I think we should remove)
```suggestion
                        firstPublished: undefined,
```